utils/build-script-impl

```diff
@@ -2628,10 +2627,32 @@ for host in "${ALL_HOSTS[@]}"; do
                     -DCMAKE_INSTALL_LIBDIR:PATH="lib"
 
                     -DENABLE_SWIFT=YES
-                    -DSWIFT_RUNTIME_LIBDIR:PATH="${SWIFT_BUILD_PATH}/lib/swift/${SWIFT_HOST_VARIANT}/${SWIFT_HOST_VARIANT_ARCH}"
 
                     -DENABLE_TESTING=YES
                   )
+
+                  case ${product} in
+                    libdispatch)
+                      cmake_options=(
+                        ${cmake_options[@]}
+                        -DSWIFT_RUNTIME_LIBDIR:PATH="${SWIFT_BUILD_PATH}/lib/swift/${SWIFT_HOST_VARIANT}/${SWIFT_HOST_VARIANT_ARCH}"
+                        -DBUILD_SHARED_LIBS=YES
+                      )
+                    ;;
+                    libdispatch_static)
+                      cmake_options=(
+                        ${cmake_options[@]}
+                        -DBUILD_SHARED_LIBS=NO
+                        -DSWIFT_RUNTIME_LIBDIR:PATH="${SWIFT_BUILD_PATH}/lib/swift_static/${SWIFT_HOST_VARIANT}/${SWIFT_HOST_VARIANT_ARCH}"
+                        -DINSTALL_TARGET_DIR:PATH=lib/swift_static/${target}
+                      )
```

Comment:
Both `SWIFT_RUNTIME_LIBDIR` and `INSTALL_TARGET_DIR` should no longer be needed with https://github.com/apple/swift/pull/19834.

Reply:
Thanks [~compnerd], I'll take a look at this and rebase next week (unless this is blocking your work in which case feel free to take it over).